sources/packages/backend/libs/integrations/src/services/restriction/student-restriction.service.ts

```diff
@@ -0,0 +1,172 @@
+import { Injectable } from "@nestjs/common";
+import { NotificationActionsService } from "@sims/services/notifications";
+import {
+  RecordDataModelService,
+  RestrictionActionType,
+  StudentRestriction,
+} from "@sims/sims-db";
+import {
+  ArrayContains,
+  DataSource,
+  EntityManager,
+  In,
+  Not,
+  SelectQueryBuilder,
+} from "typeorm";
+export const RESTRICTION_NOT_ACTIVE = "RESTRICTION_NOT_ACTIVE";
+export const RESTRICTION_NOT_PROVINCIAL = "RESTRICTION_NOT_PROVINCIAL";
+/**
+ * While performing a possible huge amount of select,
+ * breaks the execution in chunks.
+ */
+const NOTIFICATIONS_SELECT_CHUNK_SIZE = 1000;
+
+/**
+ * Service layer for Student Restriction.
+ */
+@Injectable()
+export class IntegrationStudentRestrictionService extends RecordDataModelService<StudentRestriction> {
+  constructor(
+    private readonly dataSource: DataSource,
```

Comment:
Please change it to `dataSource: DataSource` removing the `private readonly`.

Reply:
removing only `private`. when `readonly` was removed, it was throwing error, 
![image](https://user-images.githubusercontent.com/77353155/209401029-f70fe169-480f-4c92-888e-2ea171dc349e.png)